halo2_proofs/src/poly/domain.rs

```diff
@@ -555,3 +669,25 @@ fn test_l_i() {
         assert_eq!(eval_polynomial(&l[(8 - i) % 8][..], x), evaluations[7 - i]);
     }
 }
+
+#[test]
```

Comment:
It will great if we add an performance comparison between this new approach with the old $m*n$-sized cosetFFT to see there is performance degenerate.